### PR TITLE
imxrt117x: Add cache management to platformctl

### PR DESCRIPTION
**Pull request by agkaminski**
DONE: RTOS-714

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [x] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by ziemleszcz — DISMISSED** [review]
LGTM

**Review by gerard5 — COMMENTED** [review]
Similar change for `hal_platformctl()` can be easily applied for `rt10xx` family.

File: hal/armv7m/cpu.c [hunk 1]
```diff
@@ -315,8 +315,7 @@ void hal_cleanDCache(ptr_t start, size_t len)
 	(void)len;
 
 #ifdef CPU_IMXRT
-	/* Clean the whole cache, no harm in being overzealous */
-	_imxrt_cleanDCache();
+	_imxrt_cleaninvalDCacheAddr((void *)start, len);
```
Comment (gerard5) [inline hal/armv7m/cpu.c:318]: nit: this change removes `_imxrt_cleanDCache();`,  and add `_imxrt_cleaninvalDCacheAddr((void *)start, len);`, so the above
```
	(void)start;
	(void)len;
```
can be moved to `#else` block where arguments are really unused.

Comment (agkaminski) [reply hal/armv7m/cpu.c:315]: Oops, thanks

File: include/arch/imxrt1170.h [hunk 2]
```diff
@@ -308,7 +308,7 @@ enum { cti0_err_irq = 17 + 16, cti1_err_irq, core_irq, lpuart1_irq, lpuart2_irq,
 typedef struct {
 	enum { pctl_set = 0, pctl_get } action;
 	enum { pctl_devclock = 0, pctl_iogpr, pctl_iolpsrgpr, pctl_iomux, pctl_iopad, pctl_ioisel, pctl_reboot, pctl_devcache,
-		pctl_lpcg } type;
+		pctl_lpcg, pctl_cleanInvalDCache } type;
```
Comment (ziemleszcz) [inline include/arch/imxrt1170.h:311]: Maybe rename to `pctl_clean_inval_dcache`.

File: hal/armv7m/imxrt/117x/imxrt117x.h [hunk 3]
```diff
@@ -26,7 +26,7 @@ extern int hal_platformctl(void *ptr);
 extern unsigned int _imxrt_cpuid(void);
 
 
-extern void _imxrt_cleanDCache(void);
+extern void _imxrt_cleaninvalDCacheAddr(void *addr, u32 sz);
```
Comment (ziemleszcz) [inline hal/armv7m/imxrt/117x/imxrt117x.h:29]: Rename to `_imxrt_cleanInvalDCacheAddr`.

File: include/arch/imxrt1170.h (hunk 2 above)
Comment (agkaminski) [reply include/arch/imxrt1170.h:311]: It should be camelCase

File: hal/armv7m/imxrt/117x/imxrt117x.h (hunk 3 above)
Comment (agkaminski) [reply hal/armv7m/imxrt/117x/imxrt117x.h]: Done

**Review by ziemleszcz — APPROVED** [review]

**Pull request merged**